### ISSUE TITLE
[Bug] <All ID(or maybe more) ZippyShareScarpingExt>: getting video link failed

### ISSUE THREAD

**Issue by MoonLGH** [jmir1/aniyomi-extensions#200]
**PLEASE READ THIS**

---

## Device information
- Tachiyomi version:  v0.12.3.5
- Android version: 11

## Source information
- Name: All Zippyshare scraping Extenstion(tested on otakudesu,oploverz,neonime)
- Extension version: OD7,OL7,NN2

## Steps to reproduce
1. Use extensions from list above
2. Try watch something

### Expected behavior
Episode link getted from the zippyshare link

### Actual behavior
Episode failed to get video link

## Other details
Sus part
looking at the zippyshare html script tag
```
    var n = 911786%2;
    var b = 911786%3;
    var z = 911789;
    document.getElementById('dlbutton').href = "/d/jWUcqz5U/"+(n + b + z - 3)+"/Otakudesu.vip_SnNY--11_360p.mp4";
    if (document.getElementById('fimage')) {
        document.getElementById('fimage').href = "/i/jWUcqz5U/"+(n + b + z - 3)+"/Otakudesu.vip_SnNY--11_360p.mp4";
    }
```
and the code that uses to scrape it, 
```
        val res = client.newCall(GET(element.attr("href"))).execute().asJsoup()
        val scr = res.select("script:containsData(dlbutton)").html()
        var url = element.attr("href").substringBefore("/v/")
        val firstString = scr.substringAfter(" = \"").substringBefore("\" + ")
        val num1 = scr.substringAfter("+ (").substringBefore(" % ").toInt()
        val num2 = scr.substringAfter(" % ").substringBefore(" + ").toInt()
        val num4 = scr.substringAfter(" % ").substringBefore(") + ").substringAfter(" % ").toInt()
        val lastString = scr.substringAfter(") + \"").substringBefore("\";")
        val num = (num1 % num2) + (num1 % num4)
        url += firstString + num.toString() + lastString
        val quality = with(url) {
            when {
                contains("1080p") -> "1080p"
                contains("720p") -> "720p"
                contains("480p") -> "480p"
                contains("360p") -> "360p"
                else -> "Default"
            }
        }
```
i think the the calculation part is changed to variable ?

**Comment by MoonLGH** [comment]
another simple way to fix it is just spawn request, wait for like 3 seconds and scrape the dlbutton href 👍

**Comment by Luft21** [comment]
check [this](https://github.com/jmir1/aniyomi-extensions/pull/199/commits/913768ffa8aefff5d6cdfe5673100d917ccf44bf)

**Comment by MoonLGH** [comment]
But not only od that use zippyshare though.. i guess im do it, anyways thank you for the snippet

**Comment by MoonLGH** [comment]
oh yea, fixed in #199 #201
